### PR TITLE
Intimidate and Super Sweet Sugar prints the right message and doesn't play animation when already at -6

### LINKED ISSUE #4401: Intimidate plays stat stage decrease animation and prints message when already at -6
### Description

https://github.com/rh-hideout/pokeemerald-expansion/assets/56992013/a2882ab0-1924-40b6-aef3-f5b8f505a9d9

### Version

1.8.2 (Latest release)

### Upcoming/master Version

_No response_

### Discord contact info

duke5614

### PR DESCRIPTION
**Pull request by Sneed69**
## Description
When attack or evasion can't be further reduced by these abilities, it now prints "Pokemon's stat won't go lower!"

Tests by @AlexOn1ine 

## Issue(s) that this PR fixes
Fixes #4401

## **Discord contact info**
duke5614

### CODE REVIEW

**Comment by Bassoonian** [comment]
Can we get tests for this?

**Comment by AlexOn1ine** [comment]
Could this maybe solved in the switch in ability check?
```C
        case ABILITY_INTIMIDATE:
            if (!gSpecialStatuses[battler].switchInAbilityDone)
            {
                gBattlerAttacker = battler;
                gSpecialStatuses[battler].switchInAbilityDone = TRUE;
                SET_STATCHANGER(STAT_ATK, 1, TRUE);
                BattleScriptPushCursorAndCallback(BattleScript_IntimidateActivates);
                effect++;
            }
            break;
```

**Comment by AlexOn1ine** [comment]
> Could this maybe solved in the switch in ability check? Also I would assume same goes for Supersweet Syrup
> 
> ```c
>         case ABILITY_INTIMIDATE:
>             if (!gSpecialStatuses[battler].switchInAbilityDone)
>             {
>                 gBattlerAttacker = battler;
>                 gSpecialStatuses[battler].switchInAbilityDone = TRUE;
>                 SET_STATCHANGER(STAT_ATK, 1, TRUE);
>                 BattleScriptPushCursorAndCallback(BattleScript_IntimidateActivates);
>                 effect++;
>             }
>             break;
> ```

Now that I think about your solution is probably fine because I assume there is still an ability popup even if target is at -6

**Comment by Sneed69** [comment]
Added tests by @AlexOn1ine

File: test/battle/ability/supersweet_syrup.c [hunk 1]
```diff
@@ -45,3 +45,29 @@ DOUBLE_BATTLE_TEST("Supersweet Syrup lowers evasion of both opposing mon's in ba
         EXPECT_EQ(playerRight->statStages[STAT_EVASION], DEFAULT_STAT_STAGE - 1);
     }
 }
+
+SINGLE_BATTLE_TEST("Supersweet Syrum can not further lower opponents evasion if it is at minimum stages")
```
Comment (Bassoonian) [inline test/battle/ability/supersweet_syrup.c:49]:
```suggestion
SINGLE_BATTLE_TEST("Supersweet Syrup can not further lower opponents evasion if it is at minimum stages")
```

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**